src/test/java/teammates/logic/core/FeedbackResponsesLogicTest.java

```diff
@@ -1076,6 +1077,37 @@ public void testGetSessionResultsForCourse_allQuestionsSpecificSection_shouldGen
         assertEquals(0, commentGiverVisibilityTable.size());
     }
 
+    @Test
+    public void testGetSessionResultsForCourse_splitResponseFetchByGiverAndReceiver_shouldGenerateCorrectBundle() {
+        var responseBundle = loadDataBundle("/FeedbackSessionResultsTest.json");
+        removeAndRestoreDataBundle(responseBundle);
+
+        var session = responseBundle.feedbackSessions.get("standard.session");
+
+        var instructor = responseBundle.instructors.get("instructor1OfCourse1");
+        var questionResponseMapFromMultiFetch = frLogic.getSessionResultsForCourse(
+                session.getFeedbackSessionName(), session.getCourseId(), instructor.getEmail(),
+                null, "Section A", FeedbackResultFetchType.GIVER).getQuestionResponseMap();
+        frLogic.getSessionResultsForCourse(
+                session.getFeedbackSessionName(), session.getCourseId(), instructor.getEmail(),
+                null, "Section A", FeedbackResultFetchType.RECEIVER)
+                .getQuestionResponseMap()
+                .forEach(questionResponseMapFromMultiFetch::putIfAbsent);
+
+        // Equal to session result fetch by both type
+        var questionResponseMapFromFetchBoth = frLogic.getSessionResultsForCourse(
+                session.getFeedbackSessionName(), session.getCourseId(), instructor.getEmail(),
+                null, "Section A", FeedbackResultFetchType.BOTH).getQuestionResponseMap();
+
+        for (var entry : questionResponseMapFromFetchBoth.entrySet()) {
+            var respFromFetchBoth = entry.getValue();
+            var respFromMultiFetch = questionResponseMapFromMultiFetch.get(entry.getKey());
+            assertEquals(respFromFetchBoth.size(), respFromMultiFetch.size());
+            assertTrue(new HashSet(respFromMultiFetch).equals(new HashSet(respFromFetchBoth)));
```

Comment:
This tests that all responses fetched by GIVER + all responses fetched by RECEIVER = all responses fetched by BOTH. It does have its use, but far more importantly, we need to make sure that GIVER and RECEIVER returns the correct subset of responses. Not just the number, but also the exact responses returned.